source/deployment/enterprise-deployment-guide.rst

```diff
@@ -0,0 +1,216 @@
+- Bespoke Messaging Solutions - Mattermost is designed to replace bespoke messaging solutions and provide additional `security features <https://docs.mattermost.com/overview/security.html>`_, but migrating from bespoke messengers can prove to be challenging, because the data format of such tools is unpredictable. Nonetheless we provide `multiple tools <https://docs.mattermost.com/administration/migrating.html?highlight=slack#bringing-data-from-bespoke-solutions-into-mattermost>`_ to attempt migration and have had many successful migrations with our customers.
+
+.. Note::
+    If your data in the bespoke messenger is not vital we recommend a hard switch after a period of running both systems in parallel.
+
+Single Sign-On
+--------------
+
+Mattermost can act as a `SAML 2.0 <https://docs.mattermost.com/deployment/sso-saml.html>`_ provider so setting up single sign-on is a straightforward matter.
+
+We support these SSO-Services:
+
+- `OneLogin <https://docs.mattermost.com/deployment/sso-saml-onelogin.html>`_
+- `Okta <https://docs.mattermost.com/deployment/sso-saml-okta.html>`_
+- `GitLab <https://docs.mattermost.com/deployment/sso-gitlab.html>`_
+- `Google People API <https://docs.mattermost.com/deployment/sso-google.html>`_
+- `AD/LDAP <https://docs.mattermost.com/deployment/sso-ldap.html>`_
+- `Azure Active Directory and Office 365 <https://docs.mattermost.com/deployment/sso-office.html>`_
+- `Microsoft ADFS <https://docs.mattermost.com/deployment/sso-saml-adfs-msws2016.html>`_
+
+Compliance
+----------
+
+When you have to meet compliance requirements - especially when working with proxies - make sure to plan ahead to avoid infrastructure redesign while deploying Mattermost. Here is how Mattermost supports your compliance needs:
+
+- Outbound Proxy - In some scenarios, like monitoring outbound traffic or controlling which websites can appear in link previews, you may wish to `use Mattermost behind a proxy <https://docs.mattermost.com/install/outbound-proxy.html>`_
+- Electronic Discovery - Electronic Discovery (eDiscovery) is the process of searching electronic data to be used as evidence in a legal case. We have put together the `eDiscovery documentation <https://docs.mattermost.com/administration/ediscovery.html>`_ to help.
+- Compliance Export - This feature enables `compliance exports <https://docs.mattermost.com/administration/compliance-export.html>`_ to be produced from the System Console, containing all messages
+- Data Retention - By default, Mattermost provides unlimited search history storing all messages without an expiration date. These defaults can be `changed by setting Message Retention and File Retention <https://docs.mattermost.com/administration/data-retention.html>`_ to a specific duration in the System Console.
+- Custom Terms of Service - If your Organization requires the use of `custom ToS <https://docs.mattermost.com/administration/custom-terms-of-service.html>`_, this can be done in the Mattermost System Console.
```

Comment:
```suggestion
- Custom Terms of Service - If your organization requires the use of `custom ToS <https://docs.mattermost.com/administration/custom-terms-of-service.html>`_, this can be done in the Mattermost System Console
```